### DISCUSSION TITLE
Cant Remove white background

### DISCUSSION THREAD

**Discussion by CroSSer23** [ImageMagick/ImageMagick#6791] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows 11

### Description

`magick convert 04.png -fuzz 5% -transparent white result.png` - my command
Stock photo
![04](https://github.com/ImageMagick/ImageMagick/assets/25705091/da0349b0-3c19-4781-b36c-6008fa64437b)

fuzz - 0%
![result_fuzz0](https://github.com/ImageMagick/ImageMagick/assets/25705091/67db0f72-eef3-43d6-8904-dbe3af1990cf)

fuzz - 5%
![result_fuzz5](https://github.com/ImageMagick/ImageMagick/assets/25705091/555c8788-1063-4c80-923c-892fc3716ebf)

fuzz - 10%
![result_fuzz10](https://github.com/ImageMagick/ImageMagick/assets/25705091/2490596d-73c7-45f3-87b6-04321104bdbf)

photoshop manual removing
![photoshop](https://github.com/ImageMagick/ImageMagick/assets/25705091/18911c09-0e53-48a3-be00-ecd6a5843c40)

how do you get that result?

**Comment by snibgo** [comment]
Your IM command makes white pixels transparent. But that isn't what you want, because the white in the speech bubbles should stay white.

A flood fill from the top-left pixel does what you want:
```
magick thisHerb.png -fuzz 50% -fill Transparent -draw "color 0,0 floodfill" out.png
```

**Reply by CroSSer23** [reply]
Wow, this solution is good, but there are a few pictures that don't process well
Before
![02](https://github.com/ImageMagick/ImageMagick/assets/25705091/a7fae6fc-3f90-4531-8fec-e7b302faf8c8)
After
![02](https://github.com/ImageMagick/ImageMagick/assets/25705091/56317d48-12d0-4321-a5cd-9a9f1188f19c)

Before
![05](https://github.com/ImageMagick/ImageMagick/assets/25705091/e77c70a0-0f22-441b-8e2d-6a1b50fd676c)
After
![05](https://github.com/ImageMagick/ImageMagick/assets/25705091/02b5b9d2-80a0-48a7-8c46-459218f14729)

**Comment by snibgo** [comment]
In your first example ("Hmph! This herb..."), the speech bubbles have black boundaries, which protects them from the flood-fill.

The third example ("Don't worry.") has black lines that extend to the image edge. Doing a flood-fill from each corner easily fixes that.
```
magick dontWorry.png -fuzz 50% -fill Transparent -draw "color 0,0 floodfill color %[fx:w-1],0 floodfill color 0,%[fx:h-1] floodfill color %[fx:w-1],%[fx:h-1] floodfill" x.png
```

Your second example ("Am I really going to marry this person?") does not have black boundaries, so a flood-fill from corners will spread into areas we don't want to change. One method is to process the image to create a mask, for example by spreading the black marks. Sorry, I don't have time right now to work out the details.

**Reply by CroSSer23** [reply]
You're a master of your craft, the solution is quite satisfying, but can you make it so that these artefacts don't happen?
I replaced the fuzz with 3% and it looks quite normal
![x](https://github.com/ImageMagick/ImageMagick/assets/25705091/408bca73-b844-4ed0-bf5a-33dcd3b696e2)

**Comment by fmw42** [comment]
Replace "color" with alpha and convert with magick.

magick image.png -fill none -draw "alpha 0,0 floodfill" x.png

![x](https://github.com/ImageMagick/ImageMagick/assets/13458218/45054e74-2728-4f8c-b7ff-c2cd5c858b5f)

Download x.png to see transparent area

**Reply by CroSSer23** [reply]
I don't really understand your solution

**Reply by fmw42** [reply]
I am just showing your the correct format for flood filling with transparency. I only did one of your commands. As I said in your command change convert to magick and color to alpha.  See https://imagemagick.org/Usage/draw/#alpha

**Comment by snibgo** [comment]
An alternative to floodfilling from all corners: we can add a thin white border, then floodfill from any edge pixel, then remove the pixels added by the border.
```
magick amIReally.png -bordercolor White -border 1 -fuzz 3% -fill Transparent -draw "color 0,0 floodfill" -shave 1x1 x.png
```
For the "Am I Really..." image, there is no obviously correct solution.
